### PR TITLE
Fix arguments for RZXCalibrationBuilder and EchoRZXWeylDecomposition

### PR DESCRIPTION
**Pull request by mtreinish**
### Summary

The RZXCalibrationBuilder and EchoRZXWeylDecomposition transpiler passes
were previously taking BaseBackend instances as arguments. Besides that
being a deprecated class which will be removed soon, it also is
incorrect because backend objects are not guaranteed to be pickleable so
when a pass manager runs in parallel processes this will cause an error.
This was never caught because these passes aren't part of any default
pass managers and their tests don't include running them as part of
transpile().

To fix this passes typically take the properties of a backend they
require. This is being reworked for BackendV2 in #5885 so a target
object can be used to encapsulate the model of a backend so we have a
single data structure to pass around, but until that is the minimum
backend version we need to also support taking the individual components
for BackendV1 and BaseBackend.

This commit changes the pass constructors to take only use the parameters
from the backend object used in the pass internals and stop requiring a
backend object be used to construct an instance of either pass.

### Details and comments

### CODE REVIEW

File: qiskit/transpiler/passes/calibration/builders.py [hunk 1]
```diff
@@ -103,26 +103,32 @@ class RZXCalibrationBuilder(CalibrationBuilder):
...
         Raises:
             QiskitError: if open pulse is not supported by the backend.
         """
         super().__init__()
-        if not backend.configuration().open_pulse:
+        if instruction_schedule_map is None or qubit_channel_mapping is None:
             raise QiskitError(
                 "Calibrations can only be added to Pulse-enabled backends, "
                 "but {} is not enabled with Pulse.".format(backend.name())
             )
```
Comment (jakelishman) [inline qiskit/transpiler/passes/calibration/builders.py:128]: name error on `backend` in the error

File: qiskit/transpiler/passes/calibration/builders.py [hunk 2]
```diff
@@ -103,26 +103,32 @@ class RZXCalibrationBuilder(CalibrationBuilder):
     angle. Additional details can be found in https://arxiv.org/abs/2012.11660.
     """
 
-    def __init__(self, backend: basebackend):
```
Comment (jakelishman) [inline qiskit/transpiler/passes/calibration/builders.py:106]: The `from qiskit.providers import basebackend` import up at the top will be unused now.

File: qiskit/transpiler/passes/calibration/builders.py (hunk 1 above)
Comment (mtreinish) [reply qiskit/transpiler/passes/calibration/builders.py]: Oops I pushed before committing my fixes from lint. It's updated now

File: qiskit/transpiler/passes/calibration/builders.py [hunk 3]
```diff
@@ -103,26 +102,29 @@ class RZXCalibrationBuilder(CalibrationBuilder):
...
+    def __init__(
+        self,
+        instruction_schedule_map: InstructionScheduleMap,
+        qubit_channel_mapping: List[List[str]],
+    ):
```
Comment (jakelishman) [inline qiskit/transpiler/passes/calibration/builders.py:109]: Technically this is a public API breaking change. Are we ok to do this?  At the very least, we'll need an upgrade note.

Comment (mtreinish) [reply qiskit/transpiler/passes/calibration/builders.py:109]: Has this pass been in a release? I looked at the git log and it showed the first commit at 08/31 which was after the 0.18 release

Comment (jakelishman) [reply qiskit/transpiler/passes/calibration/builders.py:109]: Yes - it was in #5752 which was in 0.17.  Its import path and a few other things have subsequently changed, but the object has been publicly released.

Comment (mtreinish) [reply qiskit/transpiler/passes/calibration/builders.py:109]: ah, ok that's what I get for not looking closely. Then let me change this PR to make the backend kwarg optional and deprecate it

Comment (mtreinish) [reply qiskit/transpiler/passes/calibration/builders.py]: Ok, it's being deprecated first in: https://github.com/Qiskit/qiskit-terra/pull/7331/commits/5225120569bf80f22106f5d3e64160b1eb491d0d

**Review by levbishop — APPROVED** [review]
LGTM

**Review by jakelishman — APPROVED** [review]

**Pull request merged**